src/lib/game/world/handler.js

```diff
@@ -128,6 +142,26 @@ class WorldHandler extends EventEmitter {
     this.renderAtCoords(player.position.x, player.position.y);
   }
 
+  addBillboardedM2(m2) {
+    this.billboardedM2s.push(m2);
+  }
+
+  animate(camera) {
+    const cameraRotated = this.prevCameraRotation === null ||
+      !this.prevCameraRotation.equals(camera.quaternion);
+
+    this.animateModels(camera, cameraRotated);
+
+    this.prevCameraRotation = camera.quaternion.clone();
+  }
+
+  animateModels(camera, cameraRotated) {
+    if (cameraRotated) {
```

Comment:
I figured we didn't want to trigger the overhead of rotating bones to billboard them in cases where the camera itself hadn't actually rotated. In practice, it might be that the billboarding needs to happen both if the camera rotates or translates, as well as if the player moves through world space (since the camera effectively follows the player).

Regardless, the idea is to avoid billboarding overhead in cases where there's no perspective change relative to objects in the world from the point of view of the camera.

Maybe a better idea would be to track all types of camera perspective changes in a single variable, and pass that along to animate?